nmdc_runtime/site/export/ncbi_xml.py

```diff
@@ -0,0 +1,431 @@
+                    "Action",
+                    children=[
+                        self.set_element(
+                            "AddFiles",
+                            attrib={"target_db": "SRA"},
+                            children=files_elements
+                            + attribute_elements
+                            + [identifier_element],
+                        ),
+                    ],
+                )
+
+                self.root.append(action)
+
+    def get_submission_xml(
+        self,
+        biosamples_list: list,
+        biosample_omics_processing_list: list,
+        biosample_data_objects_list: list,
+    ):
+        data_type = None
+        ncbi_project_id = None
+        for bsm_omprc in biosample_omics_processing_list:
+            for _, omprc_list in bsm_omprc.items():
+                for omprc in omprc_list:
+                    if "omics_type" in omprc:
+                        data_type = handle_text_value(omprc["omics_type"]).capitalize()
+
+                    if "ncbi_project_name" in omprc:
+                        ncbi_project_id = omprc["ncbi_project_name"]
+
+        self.set_description(
+            email=self.nmdc_pi_email,
+            user="National Microbiome Data Collaborative (NMDC)",
+            first=self.first_name,
+            last=self.last_name,
+            org=self.ncbi_submission_metadata.get("organization", ""),
+        )
+
+        if not ncbi_project_id:
+            self.set_bioproject(
+                title=self.nmdc_study_title,
+                project_id=ncbi_project_id,
+                description=self.nmdc_study_description,
+                data_type=data_type,
+                org=self.ncbi_submission_metadata.get("organization", ""),
+            )
+
+        self.set_biosample(
+            organism_name=self.ncbi_biosample_metadata.get("organism_name", ""),
+            org=self.ncbi_submission_metadata.get("organization", ""),
+            bioproject_id=ncbi_project_id,
+            nmdc_biosamples=biosamples_list,
+            nmdc_omics_processing=biosample_omics_processing_list,
+        )
+
+        self.set_fastq(
+            biosample_data_objects=biosample_data_objects_list,
+            bioproject_id=ncbi_project_id,
+            org=self.ncbi_submission_metadata.get("organization", ""),
+        )
+
+        rough_string = ET.tostring(self.root, "unicode")
+        reparsed = xml.dom.minidom.parseString(rough_string)
+        submission_xml = reparsed.toprettyxml(indent="    ", newl="\n")
+
+        # ============= Uncomment the following code to validate the XML against NCBI XSDs ============ #
+        # submission_xsd_url = "https://www.ncbi.nlm.nih.gov/viewvc/v1/trunk/submit/public-docs/common/submission.xsd?view=co"
+        # validate_xml(submission_xml, submission_xsd_url)
+
+        # bioproject_xsd_url = "https://www.ncbi.nlm.nih.gov/viewvc/v1/trunk/submit/public-docs/common/bioproject.xsd?view=co"
```

Comment:
This URL returns a 404?

Reply:
Great catch! I've updated it to URLs that work now, and at the moment we are not enforcing XSD validation because the we are still figuring out what the XML should look like, so we are relying on actually submitting the produced XML to a test endpoint, visualizing it through the web portal/interface and coming back to the XML with modifications.